src/main/scala/wdl4s/wdl/WdlTask.scala

```diff
@@ -45,6 +48,32 @@ object WdlTask {
   }
 
   def empty: WdlTask = new WdlTask("taskName", Seq.empty, RuntimeAttributes(Map.empty[String, WdlExpression]), Map.empty, Map.empty, null)
+  
+  def buildWomTaskDefinition(wdlTask: WdlTask): TaskDefinition = TaskDefinition(
+    wdlTask.name,
+    wdlTask.commandTemplate,
+    wdlTask.runtimeAttributes,
+    wdlTask.meta,
+    wdlTask.parameterMeta,
+    wdlTask.outputs.map(_.womOutputDefinition).toSet,
+    buildWomInputs(wdlTask),
+    buildWomDeclarations(wdlTask)
+  )
+  
+  def buildWomDeclarations(wdlTask: WdlTask): List[(String, Expression)] = wdlTask.declarations.toList collect {
+    case d if d.expression.nonEmpty =>
+      d.unqualifiedName -> PlaceholderExpression(d.wdlType)
+  }
+  
+  def buildWomInputs(wdlTask: WdlTask): Set[Callable.InputDefinition] = wdlTask.declarations collect {
+    case d if d.expression.isEmpty && !d.wdlType.isInstanceOf[WdlOptionalType] =>
+      RequiredInputDefinition(d.unqualifiedName, d.wdlType)
+    case d if d.expression.isEmpty && d.wdlType.isInstanceOf[WdlOptionalType] =>
+      OptionalInputDefinition(d.unqualifiedName, d.wdlType.asInstanceOf[WdlOptionalType])
+    case d if d.expression.nonEmpty && d.wdlType.isInstanceOf[WdlOptionalType] =>
+      OptionalInputDefinitionWithDefault(d.unqualifiedName, d.wdlType.asInstanceOf[WdlOptionalType], PlaceholderExpression(d.wdlType))
+  } toSet
```

Comment:
All three of these methods seem like they should be instance methods rather than methods on the companion object parameterized solely by an instance.